int-set/src/lib.rs

```diff
@@ -192,64 +235,98 @@ impl<T> IntSet<T> {
     }
 }
 
-impl<T: Into<u32> + Copy> FromIterator<T> for IntSet<T> {
+impl<T: IntSetMember<T>> FromIterator<T> for IntSet<T> {
     fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
         let mut s = IntSet::empty();
         s.extend(iter);
         s
     }
 }
 
-impl<T: Into<u32> + Copy> Extend<T> for IntSet<T> {
+impl<T: IntSetMember<T>> Extend<T> for IntSet<T> {
     /// Extends a collection with the contents of an iterator.
     /// This implementation is optimized to provide the best performance when the iterator contains sorted values.
     /// Consider using extend_unsorted() if the iterator is known to contain unsorted values.
     fn extend<U: IntoIterator<Item = T>>(&mut self, iter: U) {
+        let iter = iter.into_iter().map(|v| v.to_mapped());
         match &mut self.0 {
             Membership::Inclusive(s) => s.extend(iter),
             Membership::Exclusive(s) => s.remove_all(iter),
         }
     }
 }
 
-fn int_set_iter<U>(mut values: U, exclusive: bool) -> impl Iterator<Item = u32>
+fn int_set_iter<T, U>(mut values: U, exclusive: bool) -> impl Iterator<Item = T>
 where
     U: Iterator<Item = u32>,
+    T: IntSetMember<T>,
 {
     let mut cur_value = values.next();
-    let mut index: u64 = u32::MIN as u64;
+    let mut all_values = T::all_values_sorted_iter();
 
     std::iter::from_fn(move || {
         if !exclusive {
             let ret = cur_value;
             cur_value = values.next();
-            return ret;
-        }
-
-        if index > u32::MAX as u64 {
-            return None;
+            return ret.map(|v| T::from_mapped(InDomain { value: v }));
         }
 
-        while let Some(skip) = cur_value {
-            if index < skip as u64 {
+        for index in all_values.by_ref() {
+            let index = index.to_mapped();
+            loop {
+                let Some(skip) = cur_value else {
+                    // No-longer any values to skip, can freely return index
+                    return Some(T::from_mapped(InDomain { value: index }));
+                };
+
+                if index < skip {
+                    // Not a skipped value
+                    return Some(T::from_mapped(InDomain { value: index }));
+                }
+
+                cur_value = values.next();
+                if index > skip {
+                    // We've passed the skip value, need to check the next one.
+                    continue;
+                }
+
+                // index == skip, so we need to skip this index.
                 break;
             }
-
-            cur_value = values.next();
-            if index > skip as u64 {
-                continue;
-            }
-
-            index += 1;
-            continue;
         }
 
-        let next_index = index;
-        index = next_index + 1;
-        Some(next_index as u32)
+        None
     })
 }
 
+impl IntSetMember<u32> for u32 {
```

Comment:
we should also provide an impl for `GlyphId`.

Reply:
Sounds good, any other "integer" types that we use inside of fonttations that we might want to define?